### PR TITLE
2636 fsv add an export mode to be able to export final state values as csv or txt

### PR DESCRIPTION
**Pull request by barondim**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [x] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

File: dynawo/sources/API/FSV/FSVTxtExporter.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * @class TxtExporter
+ * @brief TXT exporter interface class
+ *
+ * Txt export class for final state values
+ */
+class TxtExporter : public Exporter {
+ public:
+  /**
+   * @brief Export method in TXT format. Note: Only those marked as final state
```
Comment (rosiereflo) [inline dynawo/sources/API/FSV/FSVTxtExporter.h:37]: same as above

File: dynawo/sources/API/FSV/test/TestExporters.cpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 //
-// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// Copyright (c) 2022, RTE (http://www.rte-france.com)
```
Comment (rosiereflo) [inline dynawo/sources/API/FSV/test/TestExporters.cpp:2]: I don't think the copyright should be modified

File: dynawo/sources/API/JOB/xsd/jobs.xsd [hunk 3]
```diff
@@ -171,6 +171,7 @@
 
   <xs:complexType name="FinalStateValuesEntry">
     <xs:attribute name="inputFile" use="required" type="xs:string"/>
+    <xs:attribute name="exportMode" use="required" type="dyn:FinalStateValuesExportMode"/>
```
Comment (rosiereflo) [inline dynawo/sources/API/JOB/xsd/jobs.xsd:174]: I would propose optional and with default value XML

**Review by rosiereflo — CHANGES_REQUESTED** [review]

**Comment by rosiereflo** [comment]
Also the unit tests for jobs file are missing

File: dynawo/sources/API/JOB/xsd/jobs.xsd (hunk 3 above)
Comment (gautierbureau) [reply dynawo/sources/API/JOB/xsd/jobs.xsd:174]: I would say CSV is a better default value

File: dynawo/sources/API/FSV/test/TestExporters.cpp (hunk 2 above)
Comment (barondim) [reply dynawo/sources/API/FSV/test/TestExporters.cpp:2]: This file is new tho

Comment (rosiereflo) [reply dynawo/sources/API/FSV/test/TestExporters.cpp:2]: not really, it is  a renaming of an existing file

File: dynawo/sources/API/JOB/test/TestFinalStateValuesEntry.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  boost::shared_ptr<FinalStateValuesEntry> finalStateValuesEntry = boost::shared_ptr<FinalStateValuesEntry>(new FinalStateValuesEntry());
+  // check default attributes
+  ASSERT_EQ(finalStateValuesEntry->getInputFile(), "");
+  ASSERT_EQ(finalStateValuesEntry->getExportMode(), "");
+
+  finalStateValuesEntry->setInputFile("input.fsv");
+  finalStateValuesEntry->setExportMode("XML");
+
+  ASSERT_EQ(finalStateValuesEntry->getInputFile(), "input.fsv");
+  ASSERT_EQ(finalStateValuesEntry->getExportMode(), "XML");
```
Comment (rosiereflo) [inline dynawo/sources/API/JOB/test/TestFinalStateValuesEntry.cpp:35]: Default should be CSV

Comment (barondim) [reply dynawo/sources/API/JOB/test/TestFinalStateValuesEntry.cpp:35]: This test tests the setter and the getter, not the default value.

Comment (rosiereflo) [reply dynawo/sources/API/JOB/test/TestFinalStateValuesEntry.cpp:35]: so where is the default value tested?

Comment (rosiereflo) [reply dynawo/sources/API/JOB/test/TestFinalStateValuesEntry.cpp:35]: Also, we need a test of job reading a file with an export mode in the fsv (using jobsExample.jobs I think)

**Review by rosiereflo — APPROVED** [review]

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**